### PR TITLE
3543 wmi hanging

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes #`put issue number here`.

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/exploiters/wmi/config-schema.json [hunk 1]
```diff
@@ -7,6 +7,14 @@
...
             "default": 30.0,
             "maximum": 100.0
+        },
+        "smb_connect_timeout": {
+            "title": "SMB connection timeout",
+            "description": "The maximum time (in seconds) to wait for a response on an SMB connection.",
+            "type": "number",
+            "minimum": 0,
+            "default": 15.0,
+            "maximum": 100.0
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/wmi/config-schema.json:17]: I don't think the user needs access to this. It's also confusing: The user may not know why any SMB settings are required on a WMI module.

File: monkey/agent_plugins/exploiters/wmi/src/wmi_client.py [hunk 2]
```diff
@@ -30,46 +34,57 @@ class WMIClient:
...
-            password=get_plaintext(secret_of_type(credentials, Password)),
-            domain=str(host.ip),
-            lmhash=get_plaintext(secret_of_type(credentials, LMHash)),
-            nthash=get_plaintext(secret_of_type(credentials, NTHash)),
-            oxidResolver=True,
-        )
-
+        # Impacket has a hard-coded timeout of 120 seconds
+        # so we set our own preffered one
+        # DCOMConnection.PINGTIMER = Timer(timeout, DCOMConnection.pingServer)
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/wmi/src/wmi_client.py:53]: We may just need to live with this. #3557 will prevent the agent from hanging during shutdown. We can create an issue and pursue a resolution at a later date if it's a priority. I think the only real effect, here, is that the WMI exploiter will cause propagation to be very slow in some environments.

File: monkey/agent_plugins/exploiters/wmi/src/wmi_remote_access_client.py [hunk 3]
```diff
@@ -63,8 +63,8 @@ def login(self, credentials: Credentials, tags: Set[AgentEventTag]):
         tags.update(LOGIN_TAGS)
 
         try:
-            self._wmi_client.login(self._host, credentials)
             self._smb_client.login(credentials, tags)
+            self._wmi_client.login(self._host, credentials, self._options.smb_connect_timeout)
```
Comment (mssalvatore) [inline monkey/agent_plugins/exploiters/wmi/src/wmi_remote_access_client.py:67]: Is switching this order intentional?

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/agent_plugins/exploiters/wmi/src/wmi_remote_access_client.py (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/agent_plugins/exploiters/wmi/src/wmi_remote_access_client.py:67]: Yep, I saw everywhere they were doing it like this. So I thought maybe it matters but nope. I will get it back to the original order

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**